cs/API-v2/Program.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.IO;
+using System.Net.Http;
+using System.Security.Cryptography;
+using System.Text;
+using Org.BouncyCastle.Crypto;
+using Org.BouncyCastle.Crypto.Parameters;
+using Org.BouncyCastle.OpenSsl;
+using Org.BouncyCastle.Security;
+
+namespace WalutomatApiV2Example
+{
+    class Program
+    {
+        static string host = "https://api.walutomat.pl";
+        static void Main(string[] args)
+        {
+            switch (args[0])
+            {
+                case "orders-active":
+                    getActiveOrders();
+                    break;
+                case "wallet":
+                    getWalletBalance();
+                    break;
+                case "order-create":
+                    createOrder();
+                    break;
+                case "order-withdraw":
+                    withdrawOrder(args[1]);
+                    break;
+                default:
+                    Console.WriteLine("Run the program with any of these arguments: \n* orders-active, \n* wallet, \n* order-create, \n* order-withdraw");
+                    break;
+            }
+        }
+
+        static void getActiveOrders()
+        {
+            string now = DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ssZ");
+            string endpoint = "/api/v2.0.0/market_fx/orders/active";
+            string stringToSign = now + endpoint;
+
+            var http = new HttpClient();
+            var req = new HttpRequestMessage(HttpMethod.Get, host + endpoint);
+            req.Headers.Add("X-API-Key", getApiKey());
+            req.Headers.Add("X-API-Signature", sign(stringToSign));
+            req.Headers.Add("X-API-Timestamp", now);
+
+            var res = http.SendAsync(req).Result;
+            Console.WriteLine(res.Content.ReadAsStringAsync().Result);
+            return;
+        }
+
+        static void createOrder()
+        {
+            string submitId = Guid.NewGuid().ToString();
+            string now = DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ssZ");
+            string endpoint = "/api/v2.0.0/market_fx/orders";
+            string body = $"currencyPair=EURPLN&buySell=BUY&volume=90.00&volumeCurrency=EUR&limitPrice=4.2456&dryRun=false&submitId={submitId}";

```

Comment:
suggestion: use proper `x-www-form-urlencoded` encoder if it's not too much trouble…

Reply:
I'd definitely do it if this code was meant to sent parametrized payload in a real life scenario. On the contrary, it sends only one payload for demonstrative purposes.

Reply:
Since this is an example I would expect to copy+paste it to my real-life code and be ready to go :grinning:

Reply:
Ok, I did my totally best.